.github/workflows/ontology_diff.yml

```diff
@@ -0,0 +1,135 @@
+name: 'Create OAK diffs on Pull requests'
+
+on:
+    # Triggers the workflow on pull request events for the master branch
+    pull_request:
+        branches: [ master ]
+
+    # Allows you to run this workflow manually from the Actions tab
+    workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build_branch:
+    runs-on: ubuntu-latest
+    container: obolibrary/odkfull:v1.5
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build ontology
+        run: cd src/ontology; make IMP=FALSE PAT=FALSE MIR=FALSE hp.obo
+      - name: Upload tmp/hp-branch.obo
+        uses: actions/upload-artifact@v4
+        with:
+          name: hp-branch.obo
+          path: src/ontology/hp.obo
+          retention-days: 1
+  build_main:
+    runs-on: ubuntu-latest
+    container: obolibrary/odkfull:v1.5
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: master
+      - name: Classify ontology
+        run: |
+          cd src/ontology; make IMP=FALSE PAT=FALSE MIR=FALSE hp.obo
+      - name: Upload tmp/hp-main.obo
```

Comment:
Changed some of the naming to get rid of the `master` terminology which is discouraged nowadays.